pkgs/pkgs-lib/formats/libconfig/test/comprehensive/default.nix

```
@@ -0,0 +1,72 @@
+{ lib, formats, stdenvNoCC, writeText, ... }:
+let
+  libconfig = formats.libconfig { };
+
+  include_expr = {
+    val = 1;
+  };
+
+  include_file = writeText "libconfig-test-include" ''
+    val=1;
+  '';
+
+  expression = {
+    simple_top_level_attr = "1.0";
+    nested.attrset.has.a.integer.value = 100;
+    some_floaty = 29.95;
+    ## Same syntax here on these two, but they should get serialized differently:
+    # > A list may have zero or more elements, each of which can be a scalar value, an array, a group, or another list.
+    list1d = libconfig.lib.mkList [ 1 "mixed!" 5 2 ];
+    # You might also omit the mkList, as a list will be a list (in contrast to an array) by default.
+    list2d = [ 1 [ 1 1.2 "foo" ] [ "bar" 1.2 1 ] ];
+    # > An array may have zero or more elements, but the elements must all be scalar values of the same type.
+    array1d = libconfig.lib.mkArray [ 1 5 2 ];
+    array2d = [
+      (libconfig.lib.mkArray [ 1 2 ])
+      (libconfig.lib.mkArray [ 2 1 ])
+    ];
+    nasty_string = "\"@\n\\\t^*\b\f\n\0\";'''$";
+
+    weirderTypes = {
+      _includes = [ include_file ];
+      pi = 3.141592654;
+      bigint = 9223372036854775807;
+      hex = libconfig.lib.mkHex "0x1FC3";
+      octal = libconfig.lib.mkOctal "0027";
+      float = libconfig.lib.mkFloat "1.2E-3";
+      array_of_ints = libconfig.lib.mkArray [
+        (libconfig.lib.mkOctal "0732")
+        (libconfig.lib.mkHex "0xA3")
+        1234
+      ];
+      list_of_weird_types = [
+        3.141592654
+        9223372036854775807
+        (libconfig.lib.mkHex "0x1FC3")
+        (libconfig.lib.mkOctal "0027")
+        (libconfig.lib.mkFloat "1.2E-32")
```

I'd kind of expected it to be verbatim, but that's ok. What happens if it's also a valid int? That should produce something like `1.0` or `1.` I guess?

```suggestion
        (libconfig.lib.mkFloat "1.2E-32")
        (libconfig.lib.mkFloat "1")
```

---

Nice catch, rust is being weird about it and formatting the float as `1`. I'll find a workaround and add it to the test